rules/CVE-2022-3172/filter.rego

```diff
@@ -0,0 +1,33 @@
+package armo_builtins
+
+import future.keywords.in
+
+deny[msg] {
+	# find aggregated API APIServices
+	obj = input[_]
+	obj.apiVersion == "apiregistration.k8s.io/v1"
+	obj.kind == "APIService"
+	api_service := obj.spec.service
+
+	# check API server version vulnerability
+	api_infos = [api_info |
+		api_info := input[i]
+		api_info.apiVersion == "apiserverinfo.kubescape.cloud/v1beta0"
+		api_info.kind == "APIServerInfo"
+		api_info.metadata.name == "version"
+	]
+
+	# Find the service that exposes the extended API
+	services = [ obj |
+		obj := input[j]
+		obj.apiVersion == "v1"
+		obj.kind == "Service"
+		obj.metadata.name == api_service.name
+	]
+
+
+	msg := {
+		"alertObject": {"k8sApiObjects": [obj]},
```

Comment:
please change to externalObjects as in raw.rego

Reply:
The same as above